services/network/HttpClientBasic.cpp

```diff
@@ -73,7 +73,7 @@ namespace services
     {
         auto path = services::PathFromUrl(url);
         auto offset = path.empty() ? url.size() : path.begin() - url.begin();
-        return infra::BoundedString(infra::MakeRange(url.begin() + offset, url.begin() + url.max_size() - offset), path.size());
+        return infra::BoundedString(infra::MakeRange(url.begin() + offset, url.begin() + url.max_size()), path.size());
```

Comment:
Good find! Can you also add one or a few microtests on the HttpClientBasic::Path() function that verifies the correct resolution of the bug?

Reply:
I have added 2 tests
- one that checks whether Path() returns the expected value
- one that checks whether the returned range has the expected size